Examples/ObjcExample/ObjcExample/ViewController.m

```diff
@@ -12,117 +12,128 @@ @implementation ViewController
 - (void)viewDidLoad {
   [super viewDidLoad];
   // Do any additional setup after loading the view, typically from a nib.
+}
 
-  self.trackingIsAllowed = false;
++ (void)useIDFA:(void (^)(bool, NSString *))completionHandler {
   if (@available(iOS 14, *)) {
     [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(
                            ATTrackingManagerAuthorizationStatus status) {
-      self.trackingIsAllowed = status == ATTrackingManagerAuthorizationStatusAuthorized;
-      if (self.trackingIsAllowed) {
-        self.idfa = [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
+      bool isTrackingEnabled = status == ATTrackingManagerAuthorizationStatusAuthorized;
+      NSString *idfa = nil;
+      if (isTrackingEnabled) {
+        idfa = [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
       }
+      completionHandler(isTrackingEnabled, idfa);
```

Comment:
Can we simplify this and avoid calling the hook when the IDFA is not available? That would avoid those `if (isTrackingEnabled)` on each method, and I think we can fill the label in right here instead, in this method.

Reply:
I would prefer not to, since this serves as example, it's preferable to be as generic as possible. This function could be in a separate class where it would not have access to the label

Reply:
I am ok with that.

However, I think that if we are looking at these as something that everyone will use as is, we should probably provide it as part of the SDK (I am not suggesting that).